### PR TITLE
[AIRFLOW-3439] Decode logs with  'utf-8'

### PR DESCRIPTION
**Pull request by RasPavel**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

**Comment by XD-DENG** [comment]
The default encoding of `.decode()` is already "utf-8". Please refer to https://docs.python.org/3/library/stdtypes.html#bytes.decode

So `.decode('utf-8')` is no difference from `.decode()`.

File: airflow/utils/log/gcs_task_handler.py [hunk 1]
```diff
@@ -129,7 +129,7 @@ def gcs_read(self, remote_log_location):
         :type remote_log_location: str (path)
         """
         bkt, blob = self.parse_gcs_url(remote_log_location)
-        return self.hook.download(bkt, blob).decode()
+        return self.hook.download(bkt, blob).decode('utf-8')
```
Comment (XD-DENG) [inline airflow/utils/log/gcs_task_handler.py:132]: The default encoding of `.decode()` is already "utf-8". Please refer to https://docs.python.org/3/library/stdtypes.html#bytes.decode

So `.decode('utf-8')` is no difference from `.decode()`.

**Review by XD-DENG — COMMENTED** [review]
This code change is not changing anything actually. Please refer to my line comment.

Please correct me if I'm wrong.

File: airflow/utils/log/gcs_task_handler.py (hunk 1 above)
Comment (mik-laj) [reply airflow/utils/log/gcs_task_handler.py:132]: > So .decode('utf-8') is no difference from .decode().

I have a different opinion.  Airflow support Python 2.7, 3.6=> ([Source](https://github.com/apache/airflow/blob/master/setup.py#L424))

In documentation for Python 2.7, you can read a fragment:
> Python’s default encoding is the ‘ascii’ encoding.
([Source](https://docs.python.org/2/howto/unicode.html))

It is also worth quoting another fragment 
> str.decode([encoding[, errors]])
> Decodes the string using the codec registered for encoding. encoding defaults to the default string encoding.
([Source]( https://docs.python.org/2.7/library/stdtypes.html#str.decode))

Taking into account the quotations above, the change proposed here changes the behavior of the program.

I hope that the explanations are sufficient and clear.

Comment (RasPavel) [reply airflow/utils/log/gcs_task_handler.py:132]: Exactly, I get encoding error with python 2.7, forgot to mention that

Comment (XD-DENG) [reply airflow/utils/log/gcs_task_handler.py:132]: I see. Thanks @mik-laj and @RasPavel for the clarification

**Review by mik-laj — APPROVED** [review]

**Comment by feng-tao** [comment]
lgtm

**Comment by acroos** [comment]
This actually introduces a bug when using the `json-file` log driver.  The [pull method](https://docker-py.readthedocs.io/en/stable/api.html#docker.api.image.ImageApiMixin.pull) on `APIClient` returns a generator, which in this case returns `dict`s. The result is the following error:
```
[2020-04-29 11:14:59,010] {taskinstance.py:1145} ERROR - 'dict' object has no attribute 'decode'
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 983, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/usr/local/lib/python3.7/site-packages/airflow/operators/docker_operator.py", line 269, in execute
    output = json.loads(l.decode('utf-8').strip())
AttributeError: 'dict' object has no attribute 'decode'
```

**Pull request merged**